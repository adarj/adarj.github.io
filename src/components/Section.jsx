import React from 'react';
import Hero from 'react-lazy-hero';
import styled from 'styled-components';
import tw from 'tailwind.macro';

const Heading = styled.h1`${tw`m-20`};`;
const Text = styled.p`${tw`m-20`};`;

const Section = (props) => (
  <Hero opacity={0.0} minHeight="100vh">
    <Heading data-cy="about-me-heading">{props.heading}</Heading>
    <Text data-cy="about-me-text">{props.text}</Text>
  </Hero>
);

export default Section;