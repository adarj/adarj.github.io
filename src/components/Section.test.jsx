import React from 'react';
import { shallow } from 'enzyme';

import Section from './Section';

describe('Section', () => {
  it('renders correctly', () => {
    const tree = shallow(<Section heading="Test Section" text="lorem ipsum." />);
    expect(tree).toMatchSnapshot();
  });
});
