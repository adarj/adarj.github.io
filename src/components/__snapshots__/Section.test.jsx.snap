// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section renders correctly 1`] = `
<LazyHero
  color="#fff"
  isCentered={true}
  isFixed={false}
  minHeight="100vh"
  opacity={0}
  parallaxOffset={0}
  transitionDuration={600}
  transitionTimingFunction="ease-in-out"
>
  <styled.h1
    data-cy="about-me-heading"
  >
    Test Section
  </styled.h1>
  <styled.p
    data-cy="about-me-text"
  >
    lorem ipsum.
  </styled.p>
</LazyHero>
`;
